lib/Server.js

```diff
@@ -171,17 +171,48 @@ function Server(compiler, options) {
 				var contentBase = options.contentBase || process.cwd();
 
 				if(typeof contentBase === "object") {
-					console.log('Using contentBase as a proxy is deprecated and will be removed in the next major version. Please use the proxy option instead.\n\nTo update remove the contentBase option from webpack.config.js and add this:');
-					console.log('proxy: {\n\t"*": <your current contentBase configuration>\n}');
-					// Proxy every request to contentBase.target
-					app.all("*", function(req, res) {
-						proxy.web(req, res, contentBase, function(err) {
-							var msg = "cannot proxy to " + contentBase.target + " (" + err.message + ")";
-							this.io.sockets.emit("proxy-error", [msg]);
-							res.statusCode = 502;
-							res.end();
+					var isOldTargetRequest = contentBase.target && Object.keys(contentBase).length === 1;
+					if (isOldTargetRequest) {
+						console.log('Using contentBase as a proxy is deprecated and will be removed in the next major version. Please use the proxy option instead.\n\nTo update remove the contentBase option from webpack.config.js and add this:');
+						console.log('proxy: {\n\t"*": <your current contentBase configuration>\n}');
+						// Proxy every request to contentBase.target
+						app.all("*", function(req, res) {
+							proxy.web(req, res, contentBase, function(err) {
+								var msg = "cannot proxy to " + contentBase.target + " (" + err.message + ")";
+								this.io.sockets.emit("proxy-error", [msg]);
+								res.statusCode = 502;
+								res.end();
+							}.bind(this));
 						}.bind(this));
-					}.bind(this));
+					} else {
+						if (!Array.isArray(contentBase)) {
+							contentBase = Object.keys(contentBase).map(function (path) {
+								var options;
+								if (typeof contentBase[path] === 'string') {
+									options = {path: path, target: contentBase[path]};
+								} else {
+									options = contentBase[path];
+									options.path = path;
+								}
+								return options;
+							});
+						}
+						contentBase.forEach(function (contentItem) {
+							var isFile = false;
+							try {
+								isFile = fs.lstatSync(contentItem.target).isFile();
+							} catch (e) {
+								console.log('Warning: Serving files from '+contentItem.target+', which does not exist!');
+							}
+							if (isFile) {
+								app.get(contentItem.path, function (req, res) {
+									res.end(fs.readFileSync(contentItem.target));
```

Comment:
Is there any standard middleware (similar to the static middleware) which allows to serve a file? fs read is t simple to serve a file (header, mimetype, partial content, content-size, error handling).